contracts/ReputationMiningCycle.sol

```diff
@@ -9,9 +26,13 @@ import "./EtherRouter.sol";
 import "./ERC20Extended.sol";
 import "./ColonyNetworkStorage.sol";
 import "./IColonyNetwork.sol";
+import "./PatriciaTree/PatriciaTree.sol";
 
+// TODO: Can we handle all possible disputes regarding the very first hash that should be set?
+// Currently, at the very least, we can't handle a dispute if the very first entry is disputed.
+// A possible workaround would be to 'kick off' reputation mining with a known dummy state...
 
-contract ReputationMiningCycle {
+contract ReputationMiningCycle is PatriciaTree, DSMath {
   address colonyNetworkAddress;
   // TODO: Do we need both these mappings?
```

Comment:
I did leave a note about this todo in https://github.com/JoinColony/colonyNetwork/pull/156/files#r168525633 but it seems that the mappings have changed since. Is this still relevant?

Reply:
Looks like the names might have changed, but they're fundamentally the same. The thing that gave me pause with what you proposed there was that there would be no way to check if a miner had already submitted a hash in the current reputation cycle that was different from what they were currently proposing - we can't loop through that mapping, and even if we could gas would stop me from implementing it like that 😛 

Still can't see a way out of this. Would rather leave the TODO though to keep reminding us about it - I'm sure a good chunk of gas could be saved here.